core/src/main/java/io/undertow/server/protocol/proxy/ProxyProtocolReadListener.java

```diff
@@ -276,4 +401,19 @@ public SocketAddress getLocalAddress() {
         }
     }
 
+
+    /**
+     * This is needed to pass "freeBuffer" by reference between methods
+     */
+    private static class BooleanWrapper {
```

Comment:
I would just use AtomicBoolean for this rather than defining a whole new class

Reply:
Makes total sense... I forgot about that. Will change.